patches/tModLoader/Terraria/ID/ContentSamples.cs.patch

```diff
@@ -70,6 +70,18 @@
  			Item item = new Item();
  			item.SetDefaults(k);
  			ItemsByType[k] = item;
+ 			string name2 = ItemID.Search.GetName(item.netID);
+ 			ItemPersistentIdsByNetIds[k] = name2;
+ 			ItemNetIdsByPersistentIds[name2] = k;
++			if (!ItemID.Sets.DisableAutomaticPlaceableDrop[k]) {
++				if (item.createTile > -1)
++					TileLoader.tileTypeAndTileStyleToItemType[(item.createTile, item.placeStyle)] = item.type;
++				if (item.createWall > -1)
++					WallLoader.wallTypeToItemType[item.createWall] = item.type;
++			}
```

Comment:
I think this should go in a `FinishSetup` function in `TileLoader/WallLoader`. Is there any reason it needs to run here?

Reply:
No need, just using an existing loop, moved.